project/SparkBuild.scala

```diff
@@ -954,7 +954,12 @@ object Unsafe {
 object DockerIntegrationTests {
   // This serves to override the override specified in DependencyOverrides:
   lazy val settings = Seq(
-    dependencyOverrides += "com.google.guava" % "guava" % "33.1.0-jre"
+    dependencyOverrides += {
+      val guavaVersion =
+        SbtPomKeys.effectivePom.value.getProperties.get(
+          "guava.version").asInstanceOf[String]
+      "com.google.guava" % "guava" % guavaVersion
```

Comment:
Just out of curiosity, how will the Guava version be resolved without this override?